torch/csrc/jit/codegen/cuda/lower_validation.cpp

```diff
@@ -288,6 +288,13 @@ class VectorizeValidator : public OptInDispatch {
     domains_.insert(m->inner());
   }
 
+  void handle(Swizzle2D* swizzle) final {
+    if (swizzle->outX() == vectorized_id_ || swizzle->inX() == vectorized_id_) {
```

Comment:
Good catch. I think the swizzle type needs to be checked to decide which axes to check. Will add it.

Reply:
I feel that both X and Y should be checked for all swizzles. We can vectorize a dimension only if it is contiguous. And the only case where there is one dimension is contiguous with swizzle is that the swizzle is effectively a 1D swizzle. That is, there is a single-variable function `f` such that `mySwizzle(x, y) = f(x), y` or `mySwizzle(x, y) = x, f(y)`, but I don't think we have such swizzle. And if in the future, we do need such thing, we should make a new class `Swizzle1D` to model it, it should be a `Swizzle2D`.

As an example, we can consider the Xor swizzle:
```
//    1   2  3  4       1   2   3  4
//    5   6  7  8       6   5   8  7
//    9  10 11 12  =>   11  12  9 10
//    13 14 15 16       16  15 14 13
```
We can not vectorize X because in the first column, if vectorizable, we are supposed to fetch a `[1, 5, 9, 13]` but we are fetching `[1, 6, 11, 16]`. We can not vectorize Y either, because although the first row is vectorizable, the second row is not. We want to fetch `[5   6  7  8]` in this exact order, but `[6   5   8  7]` is a different order.